gr-zeromq/include/gnuradio/zeromq/pub_sink.h

```diff
@@ -57,6 +57,11 @@ namespace gr {
        */
       static sptr make(size_t itemsize, size_t vlen, char *address,
                        int timeout=100, bool pass_tags=false, int hwm=-1);
+
+      /*!
+       * \brief If the endpoint is TCP return the port number; -1 otherwise.
+       */
+      virtual int port() = 0;
```

Comment:
indeed; do you know of a reason the message variants don't inherit from the base_impl? thanks for catching the whitespace